src/services/healthcareProfessionalService.ts

```diff
@@ -134,8 +146,22 @@ Promise<Result<gqlTypes.HealthcareProfessional[]>> {
                 ? professionalsFilteredBySpokenLanguages.filter(professional =>
                     professional.degrees.some(db => filters.degrees?.some(filter => filter == db)))
                 : professionalsFilteredBySpokenLanguages ?? []
+            
+            const professionalsFilteredByName = hasNamesFilter
+                ? professionalsFilteredByDegrees.filter(professionals => {
+                    const filterName = filters.names ? filters.names[0] : false
```

Comment:
should check length otherwise 0 will be undefined when accessing
```suggestion
                    const filterName = filters.names.length ? filters.names[0] : false
```